app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CategoriesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  before_action :set_category, only: %i[show edit update destroy]
+
+  def index
+    @categories = Category.includes(:user, :transactions).order(created_at: :desc)
+  end
+
+  def show; end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
```

Comment:
Please see this error which occurs if the user does not add an icon while signing up. The app does not proceed after this error. I checked both the live version and on my machine.

![category_error](https://user-images.githubusercontent.com/35504149/190509724-62f2e5e4-2f89-431b-8db7-73d4f302a44e.png)


![live_error](https://user-images.githubusercontent.com/35504149/190509776-63de276a-aa2e-4149-8ace-13d968be33e4.png)